bitcoin/src/bip152.rs

```diff
@@ -217,14 +218,11 @@ impl HeaderAndShortIds {
                     },
                 });
             } else {
-                short_ids.push(ShortId::with_siphash_keys(
-                    &match version {
-                        1 => tx.compute_txid().to_raw_hash(),
-                        2 => tx.compute_wtxid().to_raw_hash(),
-                        _ => unreachable!(),
-                    },
-                    siphash_keys,
-                ));
+                match version {
+                    1 => short_ids.push(ShortId::with_siphash_keys(&tx.compute_txid(), siphash_keys)),
+                    2 => short_ids.push(ShortId::with_siphash_keys(&tx.compute_wtxid(), siphash_keys)),
+                    _ => unreachable!(),
```

Comment:
OT but this should be an enum. Honestly last time I looked at this module I didn't like it at all. IIUC it's a literal transaction from C++ Core code, so that's why it's not idiomatic.

Reply:
Yeah IMO all the bip modules need a bunch of love, I wasn't going to give it to them until after all the crate smashing is done.

Reply:
I made an issue